homeassistant/components/ccl/sensor.py

```diff
@@ -0,0 +1,213 @@
+    ),
+    CCLSensorTypes.CO2: SensorEntityDescription(
+        key="CO2",
+        device_class=SensorDeviceClass.CO2,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+    ),
+    CCLSensorTypes.VOLATILE: SensorEntityDescription(
+        key="VOLATILE",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS_PARTS,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
+    ),
+    CCLSensorTypes.VOC: SensorEntityDescription(
+        key="VOC",
+        translation_key="voc",
+    ),
+    CCLSensorTypes.PM10: SensorEntityDescription(
+        key="PM10",
+        device_class=SensorDeviceClass.PM10,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        translation_key="pm10",
+    ),
+    CCLSensorTypes.PM25: SensorEntityDescription(
+        key="PM25",
+        device_class=SensorDeviceClass.PM25,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        translation_key="pm25",
+    ),
+    CCLSensorTypes.AQI: SensorEntityDescription(
+        key="AQI",
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key="aqi",
+    ),
+    CCLSensorTypes.BATTERY: SensorEntityDescription(
+        key="BATTERY",
+        translation_key="battery",
+    ),
+    CCLSensorTypes.LEAKAGE: SensorEntityDescription(
+        key="LEAKAGE",
+        translation_key="leakage",
+    ),
+    CCLSensorTypes.LIGHTNING_DISTANCE: SensorEntityDescription(
+        key="LIGHTNING_DISTANCE",
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        translation_key="lightning_distance",
+    ),
+    CCLSensorTypes.LIGHTNING_DURATION: SensorEntityDescription(
+        key="LIGHTNING_DURATION",
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        translation_key="lightning_duration",
+    ),
+    CCLSensorTypes.LIGHTNING_FREQUENCY: SensorEntityDescription(
+        key="LIGHTNING_FREQUENCY",
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key="lightning_frequency",
+    ),
+    CCLSensorTypes.BATTERY_VOLTAGE: SensorEntityDescription(
+        key="BATTERY_VOLTAGE",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config entry in HA."""
+    device: CCLDevice = entry.runtime_data
```

Comment:
```suggestion
    device = entry.runtime_data
```
will be inferred once you have the custom entry type